File: arrays/219. Contains Duplicate II/containsNearbyDuplicate.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        # Dictionary to store the most recent index of each number in nums
        d = {}

        for i in range(len(nums)):
            if nums[i] in d and abs(i - d[nums[i]]) <= k:
                return True
            else:
                # Update the value with the most recent index
                d[nums[i]] = i
        
        return False

File: arrays/219. Contains Duplicate II/containsNearbyDuplicate.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    let obj = {};

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] in obj && Math.abs(i - obj[nums[i]]) <= k) {
            return true;
        } else {
            obj[nums[i]] = i;
        }
    }

    return false;
};